### PR TITLE
Refactor controller layout

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
The current controller is complex during support the duration chaos feature as I mentioned [here](https://github.com/pingcap/chaos-mesh/pull/198#issuecomment-582874084)

This request is to refactor the current layout in controller package according to @at15 's [advice](https://github.com/pingcap/chaos-mesh/pull/198#discussion_r375726244) The layout in this request would be refactored with no more redundant `chaos` packages as follows:
```sh
- reconciler
  - types.go
- podchaos
  - podfailure
    - types.go
- common
  - types.go
- twhphase
  - types.go
```

The `Reconciler` in `reconciler` package defined the basic reconciler interface and the `common` and `twophase` reconcile logic is based at the basic `reconciler`. In `podfailure` package, the `NewCommonReconciler` and `NewTwophaseRecomciler` would help to create the reconciler when either of them is needed.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: controllers/podchaos/podfailure/types.go [hunk 1]
```diff
@@ -57,13 +72,13 @@ type Reconciler struct {
 	Log logr.Logger
 }
 
-// Instance returns the instance of PodChaos
-func (r *Reconciler) Instance() *v1alpha1.PodChaos {
+// Object implements the reconciler.InnerReconciler.Object
+func (r *Reconciler) Object() reconciler.InnerObject {
```
Comment (zhouqiang-cl) [inline controllers/podchaos/podfailure/types.go]: why change to Object again?

Comment (Yisaer) [reply controllers/podchaos/podfailure/types.go]: To directly implement reconciler method (Apply / Recover / Object). The previous way is add one middle layer to transform.

**Comment by zhouqiang-cl** [comment]
@cwen0 @YangKeao @mahjonp PTAL

File: controllers/common/types.go [hunk 2]
```diff
@@ -80,6 +65,7 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			r.Log.Error(err, "failed to recover chaos")
 			return ctrl.Result{Requeue: true}, nil
 		}
+		return ctrl.Result{}, nil
```
Comment (Yisaer) [inline controllers/common/types.go]: This is and bug in common controller logic, and I fixed it in this request.

File: controllers/podchaos/types.go [hunk 3]
```diff
@@ -83,19 +85,15 @@ func (r *Reconciler) schedulePodChaos(podchaos *v1alpha1.PodChaos, req ctrl.Requ
...
-		}
-		r := twophasePodfailure.NewTwoPhaseReconciler(r.Client, r.Log.WithValues("action", "pod-failure"), req)
-		reconciler := twophase.NewReconciler(r, r.Client, r.Log)
-		return reconciler.Reconcile(req)
+		tr = podfailure.NewTwoPhaseReconciler(r.Client, r.Log.WithValues("action", "pod-kill"), req)
 	default:
-		return r.defaultResponse(podchaos), nil
+		return r.invalidActionResponse(podchaos), nil
 	}
+	reconciler := twophase.NewReconciler(tr, tr.Client, tr.Log)
```
Comment (mahjonp) [inline controllers/podchaos/types.go]: why unbox the `twophase.Reconciler` instance and box it again here?

File: controllers/podchaos/types.go [hunk 4]
```diff
@@ -17,14 +17,14 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/pingcap/chaos-mesh/controllers/podchaos/podfailure"
```
Comment (mahjonp) [inline controllers/podchaos/types.go]: group imports here

File: controllers/podchaos/types.go (hunk 3 above)
Comment (Yisaer) [reply controllers/podchaos/types.go]: updated.

File: controllers/podchaos/types.go (hunk 4 above)
Comment (Yisaer) [reply controllers/podchaos/types.go]: updated.

File: controllers/podchaos/types.go [hunk 5]
```diff
@@ -18,16 +18,14 @@ import (
 	"fmt"
 
 	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
```
Comment (cwen0) [inline controllers/podchaos/types.go]:
```
	"context"
	"fmt"

	"github.com/go-logr/logr"

	"github.com/pingcap/chaos-mesh/api/v1alpha1"
	"github.com/pingcap/chaos-mesh/controllers/common"
	"github.com/pingcap/chaos-mesh/controllers/podchaos/podfailure"
	"github.com/pingcap/chaos-mesh/controllers/podchaos/podkill"
	"github.com/pingcap/chaos-mesh/controllers/twophase"

	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/client"
```

File: controllers/twophase/types.go [hunk 6]
```diff
@@ -18,10 +18,11 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/pingcap/chaos-mesh/controllers/reconciler"
```
Comment (cwen0) [inline controllers/twophase/types.go]:
```
	"context"
	"fmt"
	"time"

	"github.com/go-logr/logr"

	"github.com/pingcap/chaos-mesh/api/v1alpha1"
	"github.com/pingcap/chaos-mesh/controllers/reconciler"
	"github.com/pingcap/chaos-mesh/pkg/utils"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/util/retry"

	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/client"
```

**Comment by Yisaer** [comment]
@cwen0 the `imports` is updated, PTAL again @mahjonp .

**Pull request merged**